pipeline-model-declarative-agent/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/agent/DeclarativeAgentDescriptor.java

```diff
@@ -0,0 +1,190 @@
+    }
+
+    /**
+     * Get a map of name-to-{@link DescribableModel} of all known/registered descriptors.
+     *
+     * @return A map of name-to-{@link DescribableModel}s
+     */
+    public static Map<String,DescribableModel> getDescribableModels() {
+        Map<String,DescribableModel> models = new HashMap<>();
+
+        for (DeclarativeAgentDescriptor d : getOrderedDescriptors()) {
+            Set<String> symbolValues = SymbolLookup.getSymbolValue(d);
+
+            if (!symbolValues.isEmpty()) {
+                models.put(symbolValues.iterator().next(), new DescribableModel<>(d.clazz));
+            }
+
+        }
+
+        return models;
+    }
+
+    /**
+     * Get the map of the subset of descriptors with single-arguments - i.e., "none" and "any".
+     * @return A map of descriptors with just one argument.
+     */
+    public static Map<String,DescribableModel> singleArgModels() {
+        Map<String,DescribableModel> models = new HashMap<>();
+
+        for (Map.Entry<String,DescribableModel> entry : getDescribableModels().entrySet()) {
+            if (entry.getValue().getParameters().isEmpty()) {
+                models.put(entry.getKey(), entry.getValue());
+            }
+        }
+
+        return models;
+    }
+
+    /**
+     * An ordered (by Extension ordinal, in descending order) list of all descriptors.
+     * @return A list of descriptors.
+     */
+    public static List<DeclarativeAgentDescriptor> getOrderedDescriptors() {
+        List<DeclarativeAgentDescriptor> orderedDescriptors = new ArrayList<>();
+
+        List<ExtensionComponent<DeclarativeAgentDescriptor>> extensionComponents = new ArrayList<>(all().getComponents());
+        Collections.sort(extensionComponents);
+
+        for (ExtensionComponent<DeclarativeAgentDescriptor> extensionComponent: extensionComponents) {
+            orderedDescriptors.add(extensionComponent.getInstance());
+        }
+
+        return orderedDescriptors;
+    }
+
+    /**
+     * An ordered list of all descriptor names.
+     *
+     * @return A list of names
+     */
+    public static List<String> getOrderedNames() {
+        List<String> orderedNames = new ArrayList<>();
+
+        for (DeclarativeAgentDescriptor d : getOrderedDescriptors()) {
+            Set<String> symbolValues = SymbolLookup.getSymbolValue(d);
+            if (symbolValues != null && !symbolValues.isEmpty()) {
+                orderedNames.add(symbolValues.iterator().next());
+            }
+        }
+
+        return orderedNames;
+    }
+
+    /**
+     * Get the descriptor for a given name or null if not found.
+     *
+     * @param name The name for the descriptor to look up
+     * @return The corresponding descriptor or null if not found.
+     */
+    @Whitelisted
+    public static @Nullable DeclarativeAgentDescriptor byName(@Nonnull String name) {
+        for (DeclarativeAgentDescriptor d : all()) {
+            if (d.getName().equals(name)) {
```

Comment:
If you delete `getName` and use symbols directly, you could simplify this to `SymbolLookup.findDescriptor`.